dgraphtest/local_cluster.go

```diff
@@ -433,3 +437,44 @@ func (c *LocalCluster) AssignUids(num uint64) error {
 	}
 	return nil
 }
+
+// CompareCommits compare given commit with current cluster's commit and return
+// true if cluster's commit is ancestor of given commit else return false
+func CompareCommits(shaToBeCompare, baseSha string) (bool, error) {
+	repo, err := git.PlainOpen(repoDir)
+	if err != nil {
+		return false, errors.Wrap(err, "error while opening git repo")
+	}
+	hash, err := repo.ResolveRevision(plumbing.Revision(shaToBeCompare))
+	if err != nil {
+		return false, errors.Wrap(err, "error while getting commit hash")
+	}
+	commitToBeCompare, err := repo.CommitObject(*hash)
+	if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("error while getting commit object of hash [%v]", hash))
+	}
+	hash, err = repo.ResolveRevision(plumbing.Revision(baseSha))
+	if err != nil {
+		return false, err
+	}
+	baseCommit, err := repo.CommitObject(*hash)
+	if err != nil {
+		return false, errors.Wrap(err, fmt.Sprintf("error while getting commit object of hash [%v]", hash))
+	}
+	isParentCommit, err := baseCommit.IsAncestor(commitToBeCompare)
+	if err != nil {
+		return false, errors.Wrap(err, "error while comparing commits")
+	}
+	return isParentCommit, nil
+}
+
+func (c *LocalCluster) SkipTest(t *testing.T, commit string) error {
```

Comment:
we probably don't need to return an error. This would be a one liner at the caller then.

Reply:
we would fail the test if errors occurs